src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntity_MassFabricator.java

```diff
@@ -230,20 +230,27 @@ public CheckRecipeResult process() {
             @Override
             protected FindRecipeResult findRecipe(GT_Recipe_Map map) {
                 if (mMode == MODE_SCRAP) {
-                    ItemStack aPotentialOutput = GT_ModHandler
-                            .getRecyclerOutput(GT_Utility.copyAmount(1, inputItems[0]), 0);
-                    GT_Recipe recipe = new GTPP_Recipe(
-                            false,
-                            new ItemStack[] { GT_Utility.copyAmount(1, inputItems[0]) },
-                            aPotentialOutput == null ? null : new ItemStack[] { aPotentialOutput },
-                            null,
-                            new int[] { 2000 },
-                            null,
-                            null,
-                            40,
-                            MaterialUtils.getVoltageForTier(1),
-                            0);
-                    return FindRecipeResult.ofSuccess(recipe);
+                    if (inputItems != null) {
+                        for (ItemStack item : inputItems) {
+                            if (item == null || item.stackSize == 0) continue;
+                            ItemStack aPotentialOutput = GT_ModHandler
+                                    .getRecyclerOutput(GT_Utility.copyAmount(1, item), 0);
+                            if (aPotentialOutput == null) continue;
```

Comment:
Isn't it supposed to consume input and output nothing, instead of just ignoring it?